File: testdata/jennies/builders/discriminator_without_option/PythonBuilder/builders/discriminator_without_option.py
import typing
from ..cog import builder as cogbuilder
from ..models import discriminator_without_option


class NoShowFieldOption(cogbuilder.Builder[discriminator_without_option.NoShowFieldOption]):
    _internal: discriminator_without_option.NoShowFieldOption

    def __init__(self):
        self._internal = discriminator_without_option.NoShowFieldOption()

    def build(self) -> discriminator_without_option.NoShowFieldOption:
        """
        Builds the object.
        """
        return self._internal    
    
    def text(self, text: str) -> typing.Self:    
        self._internal.text = text
    
        return self
    


class ShowFieldOption(cogbuilder.Builder[discriminator_without_option.ShowFieldOption]):
    _internal: discriminator_without_option.ShowFieldOption

    def __init__(self):
        self._internal = discriminator_without_option.ShowFieldOption()

    def build(self) -> discriminator_without_option.ShowFieldOption:
        """
        Builds the object.
        """
        return self._internal    
    
    def field(self, field: discriminator_without_option.AnEnum) -> typing.Self:    
        self._internal.field = field
    
        return self
    
    def text(self, text: str) -> typing.Self:    
        self._internal.text = text
    
        return self

File: testdata/jennies/builders/discriminator_without_option/RustBuilder/src/builders/discriminator_without_option.rs
use crate::cog;
use crate::types::discriminator_without_option;

#[derive(Debug, Clone)]
pub struct NoShowFieldOptionBuilder {
    internal: discriminator_without_option::NoShowFieldOption,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl NoShowFieldOptionBuilder {
    pub fn new() -> Self {
        Self {
            internal: discriminator_without_option::NoShowFieldOption::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for NoShowFieldOptionBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl NoShowFieldOptionBuilder {
    pub fn text(mut self, text: String) -> Self {
        self.internal.text = text;

        self
    }
}

impl cog::Builder<discriminator_without_option::NoShowFieldOption> for NoShowFieldOptionBuilder {
    fn build(
        &self,
    ) -> Result<discriminator_without_option::NoShowFieldOption, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}

#[derive(Debug, Clone)]
pub struct ShowFieldOptionBuilder {
    internal: discriminator_without_option::ShowFieldOption,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl ShowFieldOptionBuilder {
    pub fn new() -> Self {
        Self {
            internal: discriminator_without_option::ShowFieldOption::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for ShowFieldOptionBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl ShowFieldOptionBuilder {
    pub fn field(mut self, field: discriminator_without_option::AnEnum) -> Self {
        self.internal.field = field;

        self
    }
}

impl ShowFieldOptionBuilder {
    pub fn text(mut self, text: String) -> Self {
        self.internal.text = text;

        self
    }
}

impl cog::Builder<discriminator_without_option::ShowFieldOption> for ShowFieldOptionBuilder {
    fn build(&self) -> Result<discriminator_without_option::ShowFieldOption, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
